client/src/redux/reducers/credentialsReducer.js

```diff
@@ -28,8 +29,49 @@ const initialState = {
   }
 };
 
+const selectedIndex = function(state, credential) {
+  return _.findIndex(state.persist.selectedCredentials, nextSelected => {
+    return nextSelected.id === _.get(credential, 'id');
+  });
+};
+
 const credentialsReducer = function(state = initialState, action) {
   switch (action.type) {
+    // Persist
+    case credentialsTypes.SELECT_CREDENTIAL:
+      // Do nothing if it is already selected
+      if (selectedIndex(state, action.credential) !== -1) {
+        return state;
+      }
+
+      action.credential.selected = true;
+
```

Comment:
I think this should probably change, it's considered an anti-pattern to mutate the original objects coming in. In this case the original action object, it does work though... maybe a ```const tempActionCred = Object.assign(...``` in there

Reply:
Yeah, I just hated having to set selected value then dispatch an action to get it into the selected items list. I can check if hte item is in the selectedItems list rather than keep the flag, but that happens a lot.

Reply:
Removed the mutation and added the check in the views.